File: apps/ios/Wordocious/Sources/InviteSheet.swift
import SwiftUI
import WordociousCore
#if canImport(UIKit)
import UIKit
#endif

/// Pro-only "Invite to a VS match" sheet — ports the web InviteModal
/// (components/invites/invite-modal.tsx): pick a mode, then either generate a
/// shareable join link or send a targeted invite to a username.
struct InviteSheet: View {
    @Environment(\.dismiss) private var dismiss

    enum Tab { case link, username }

    /// The 9 VS-capable modes, single-sourced from the home catalog so each row
    /// carries its real brand icon + accent (excludes the VS card, which has no
    /// dbKey). Order matches the home grid.
    private var inviteModes: [HomeMode] { homeModes.filter { $0.dbKey != nil } }
    private var selectedHome: HomeMode { inviteModes.first { $0.dbKey == mode.rawValue } ?? inviteModes[0] }

    @State private var mode: GameMode = .duel
    @State private var modeOpen = false
    @State private var tab: Tab = .link
    @State private var username = ""
    @State private var busy = false
    @State private var error: String?
    @State private var inviteURL: String?
    @State private var sentTo: String?
    @State private var copied = false

    private var modeLabel: String { selectedHome.title }
    private let pink = Color(hex: 0xEC4899), pinkDark = Color(hex: 0xDB2777)

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                // Header — gradient title + subtitle + X close (matches web modal)
                HStack(alignment: .top) {
                    VStack(alignment: .leading, spacing: 4) {
                        Text("INVITE A FRIEND")
                            .font(Brand.font(24, .black))
                            .foregroundStyle(LinearGradient(colors: [Color(hex: 0xA78BFA), Color(hex: 0xEC4899)], startPoint: .leading, endPoint: .trailing))
                        Text("Pick a mode, then send a link or a username invite.")
                            .font(Brand.font(12, .bold)).foregroundStyle(Theme.textMuted)
                    }
                    Spacer()
                    Button { dismiss() } label: {
                        Image(systemName: "xmark")
                            .font(.system(size: 13, weight: .bold)).foregroundStyle(Theme.textMuted)
                            .frame(width: 32, height: 32)
                            .background(Circle().fill(Theme.surfaceHover))
                    }.buttonStyle(.plain)
                }

                    // Mode picker — a custom inline dropdown (styled with each
                    // mode's brand icon + accent). Expands in place and pushes the
                    // content down, instead of a floating menu that overlapped and
                    // hid the buttons underneath.
                    VStack(alignment: .leading, spacing: 6) {
                        Text("GAME MODE").font(Brand.font(10, .black)).tracking(0.6).foregroundStyle(Theme.textMuted)
                        VStack(spacing: 0) {
                            // Trigger row (shows the selected mode).
                            Button {
                                withAnimation(Theme.animation(.easeInOut(duration: 0.22))) { modeOpen.toggle() }
                            } label: {
                                HStack(spacing: 10) {
                                    ModeIconView(icon: selectedHome.icon, accent: selectedHome.accent, box: 30)
                                    Text(modeLabel).font(Brand.font(16, .black)).foregroundStyle(Theme.textPrimary)
                                    Spacer()
                                    Image(systemName: "chevron.down")
                                        .font(.system(size: 12, weight: .bold)).foregroundStyle(selectedHome.accent)
                                        .rotationEffect(.degrees(modeOpen ? 180 : 0))
                                }
                                .padding(.horizontal, 12).padding(.vertical, 10)
                                .contentShape(Rectangle())
                            }.buttonStyle(.plain)

                            // Expanded list — one styled row per mode.
                            if modeOpen {
                                Divider().overlay(Theme.border).padding(.horizontal, 8)
                                VStack(spacing: 2) {
                                    ForEach(inviteModes) { hm in modeRow(hm) }
                                }
                                .padding(.horizontal, 6).padding(.top, 4).padding(.bottom, 6)
                            }
                        }
                        .background(RoundedRectangle(cornerRadius: 14).fill(Theme.background))
                        .overlay(RoundedRectangle(cornerRadius: 14).stroke(modeOpen ? selectedHome.accent : Theme.border, lineWidth: 1.5))
                    }

                    // Tabs
                    HStack(spacing: 0) {
                        tabButton("Share link", .link)
                        tabButton("Username", .username)
                    }
                    .background(RoundedRectangle(cornerRadius: 10).fill(Theme.background))
                    .overlay(RoundedRectangle(cornerRadius: 10).stroke(Theme.border, lineWidth: 1))

                    if tab == .link { linkTab } else { usernameTab }

                    if let error {
                        Text(error).font(Brand.font(12, .bold)).foregroundStyle(Color(hex: 0xDC2626))
                    }
                }
                .padding(18)
            }
    }

    // MARK: Mode picker row

    private func modeRow(_ hm: HomeMode) -> some View {
        let selected = hm.dbKey == mode.rawValue
        return Button {
            if let m = hm.dbKey.flatMap({ GameMode(rawValue: $0) }) { mode = m; reset() }
            withAnimation(Theme.animation(.easeInOut(duration: 0.2))) { modeOpen = false }
        } label: {
            HStack(spacing: 10) {
                ModeIconView(icon: hm.icon, accent: hm.accent, box: 28)
                Text(hm.title).font(Brand.font(14, .heavy)).foregroundStyle(Theme.textPrimary)
                Spacer()
                if selected {
                    Image(systemName: "checkmark").font(.system(size: 12, weight: .black)).foregroundStyle(hm.accent)
                }
            }
            .padding(.horizontal, 8).padding(.vertical, 8)
            .frame(maxWidth: .infinity)
            .background(RoundedRectangle(cornerRadius: 10).fill(selected ? hm.accent.opacity(0.10) : Color.clear))
            .contentShape(Rectangle())
        }.buttonStyle(.plain)
    }

    // MARK: Tabs

    private func tabButton(_ title: String, _ t: Tab) -> some View {
        let active = tab == t
        return Button { tab = t; error = nil } label: {
            Text(title).font(Brand.font(13, .black))
                .foregroundStyle(active ? .white : Theme.textMuted)
                .frame(maxWidth: .infinity).padding(.vertical, 9)
                .background(RoundedRectangle(cornerRadius: 8).fill(active ? Theme.primary : .clear))
        }
        .buttonStyle(.plain).padding(2)
    }

    @ViewBuilder private var linkTab: some View {
        if let url = inviteURL {
            VStack(alignment: .leading, spacing: 10) {
                Text(url).font(Brand.font(13, .semibold)).foregroundStyle(Theme.textSecondary)
                    .lineLimit(2).truncationMode(.middle)
                    .padding(12).frame(maxWidth: .infinity, alignment: .leading)
                    .background(RoundedRectangle(cornerRadius: 10).fill(Theme.background))
                    .overlay(RoundedRectangle(cornerRadius: 10).stroke(Theme.border, lineWidth: 1))
                HStack(spacing: 10) {
                    Button { copy(url) } label: {
                        Label(copied ? "Copied!" : "Copy", systemImage: copied ? "checkmark" : "doc.on.doc")
                            .font(Brand.font(13, .black)).frame(maxWidth: .infinity).padding(.vertical, 11)
                            .background(RoundedRectangle(cornerRadius: 10).fill(Theme.surfaceHover)).foregroundStyle(Theme.textPrimary)
                    }.buttonStyle(.plain)
                    Button { share(url) } label: {
                        Label("Share", systemImage: "square.and.arrow.up")
                            .font(Brand.font(13, .black)).frame(maxWidth: .infinity).padding(.vertical, 11)
                            .foregroundStyle(.white)
                            .background(RoundedRectangle(cornerRadius: 10).fill(LinearGradient(colors: [pink, pinkDark], startPoint: .topLeading, endPoint: .bottomTrailing)))
                    }.buttonStyle(.plain)
                }
                Text("Link expires in 24 hours.")
                    .font(Brand.font(10, .bold)).foregroundStyle(Theme.textMuted)
                    .frame(maxWidth: .infinity, alignment: .center)
            }
        } else {
            Button { createLink() } label: {
                Text(busy ? "Creating…" : "Create Invite Link")
                    .font(Brand.font(14, .black)).foregroundStyle(.white)
                    .frame(maxWidth: .infinity).padding(.vertical, 13)
                    .background(RoundedRectangle(cornerRadius: 12).fill(LinearGradient(colors: [pink, pinkDark], startPoint: .topLeading, endPoint: .bottomTrailing)))
            }.buttonStyle(.plain).disabled(busy)
        }
    }

    @ViewBuilder private var usernameTab: some View {
        if let sent = sentTo {
            VStack(spacing: 8) {
                Image(systemName: "checkmark.circle.fill").font(.system(size: 32)).foregroundStyle(Color(hex: 0x22C55E))
                Text("Invite sent to @\(sent)").font(Brand.font(14, .black)).foregroundStyle(Theme.textPrimary)
                Text("They'll see it the next time they open Wordocious.").font(Brand.font(12, .bold)).foregroundStyle(Theme.textMuted).multilineTextAlignment(.center)
                Button("Send another") { reset() }.font(Brand.font(11, .bold)).foregroundStyle(Theme.textMuted).padding(.top, 4)
            }
            .frame(maxWidth: .infinity).padding(.vertical, 8)
        } else {
            VStack(alignment: .leading, spacing: 10) {
                Text("USERNAME").font(Brand.font(10, .black)).tracking(0.6).foregroundStyle(Theme.textMuted)
                TextField("e.g. wordmaster", text: $username)
                    .textInputAutocapitalization(.never).autocorrectionDisabled()
                    .padding(12).background(RoundedRectangle(cornerRadius: 10).fill(Theme.background))
                    .overlay(RoundedRectangle(cornerRadius: 10).stroke(Theme.border, lineWidth: 1))
                Button { sendToUsername() } label: {
                    Text(busy ? "Sending…" : "Send Invite")
                        .font(Brand.font(14, .black)).foregroundStyle(.white)
                        .frame(maxWidth: .infinity).padding(.vertical, 13)
                        .background(RoundedRectangle(cornerRadius: 12).fill(LinearGradient(colors: [pink, pinkDark], startPoint: .topLeading, endPoint: .bottomTrailing)))
                }.buttonStyle(.plain).disabled(busy)
            }
        }
    }

    // MARK: Actions

    private func reset() { inviteURL = nil; sentTo = nil; copied = false; error = nil }

    private func createLink() {
        busy = true; error = nil
        Task {
            let r = await InviteService.createInvite(gameMode: mode, inviteeUsername: nil)
            busy = false
            if let code = r.code { inviteURL = "https://wordocious.com/vs/join/\(code)" }
            else { error = r.error ?? "Failed to create invite" }
        }
    }

    private func sendToUsername() {
        let clean = username.trimmingCharacters(in: .whitespaces).replacingOccurrences(of: "@", with: "")
        if clean.isEmpty { error = "Enter a username"; return }
        busy = true; error = nil
        Task {
            let r = await InviteService.createInvite(gameMode: mode, inviteeUsername: clean)
            busy = false
            if r.code != nil { sentTo = clean } else { error = r.error ?? "Failed to send invite" }
        }
    }

    private func copy(_ url: String) {
        #if canImport(UIKit)
        UIPasteboard.general.string = url
        #endif
        copied = true
    }

    private func share(_ url: String) {
        ShareEvents.log(kind: "link_invite", gameMode: mode.rawValue, surface: "invite_sheet")
        #if canImport(UIKit)
        let text = "Come play me on Wordocious — \(modeLabel)."
        let items: [Any] = [text, URL(string: url) ?? url]
        let av = UIActivityViewController(activityItems: items, applicationActivities: nil)
        guard let scene = UIApplication.shared.connectedScenes.first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene,
              let root = scene.windows.first(where: { $0.isKeyWindow })?.rootViewController else { return }
        var top = root
        while let p = top.presentedViewController { top = p }
        av.popoverPresentationController?.sourceView = top.view
        av.popoverPresentationController?.sourceRect = CGRect(x: top.view.bounds.midX, y: top.view.bounds.midY, width: 0, height: 0)
        top.present(av, animated: true)
        #endif
    }
}

File: apps/android/app/src/main/kotlin/com/wordocious/app/ui/InviteSheet.kt
package com.wordocious.app.ui

import android.content.Intent
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material.icons.filled.ExpandMore
import androidx.compose.material.icons.filled.Link
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Share
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import com.wordocious.app.data.InviteService
import com.wordocious.app.ui.theme.WTheme
import kotlinx.coroutines.launch

/**
 * Pro-only "Invite a friend to a VS match" modal — Android port of the web
 * InviteModal (components/invites/invite-modal.tsx) and iOS InviteSheet: pick a
 * mode, then either generate a shareable join link or send a targeted invite to
 * a username.
 */
@Composable
fun InviteSheet(onDismiss: () -> Unit) {
    val context = LocalContext.current
    val clipboard = LocalClipboardManager.current
    val scope = rememberCoroutineScope()

    // The 9 VS-capable modes (every MODE_CARD with an engine mode — excludes the
    // "vs" card itself). Same list + brand colors/glyphs as web MODES.
    val modes = remember { MODE_CARDS.filter { it.engineMode != null } }

    var card by remember { mutableStateOf(modes.first()) }
    var modeOpen by remember { mutableStateOf(false) }
    var tab by remember { mutableStateOf("link") }
    var username by remember { mutableStateOf("") }
    var busy by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    var inviteUrl by remember { mutableStateOf<String?>(null) }
    var sentTo by remember { mutableStateOf<String?>(null) }
    var copied by remember { mutableStateOf(false) }
    val purple = Color(0xFF7C3AED)

    fun reset() { inviteUrl = null; sentTo = null; username = ""; error = null; copied = false }

    Dialog(onDismissRequest = { reset(); onDismiss() }) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(20.dp))
                .background(WTheme.surface)
                .border(1.5.dp, WTheme.border, RoundedCornerShape(20.dp))
                .padding(20.dp),
        ) {
            // Header row: gradient title + close.
            Row(verticalAlignment = Alignment.Top) {
                Column(Modifier.weight(1f)) {
                    Text(
                        "Invite a friend",
                        fontSize = 24.sp, fontWeight = FontWeight.Black,
                        style = androidx.compose.ui.text.TextStyle(
                            brush = Brush.linearGradient(listOf(Color(0xFFA78BFA), Color(0xFFEC4899))),
                        ),
                    )
                    Text(
                        "Pick a mode, then send a link or a username invite.",
                        fontSize = 12.sp, fontWeight = FontWeight.Bold, color = WTheme.textMuted,
                        modifier = Modifier.padding(top = 2.dp),
                    )
                }
                Box(
                    Modifier.size(32.dp).clip(CircleShape).background(WTheme.bg)
                        .clickable { reset(); onDismiss() },
                    contentAlignment = Alignment.Center,
                ) { Icon(Icons.Filled.Close, "Close", tint = WTheme.textMuted, modifier = Modifier.size(16.dp)) }
            }

            Spacer(Modifier.size(16.dp))

            // Tabs: Share link / Username.
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.fillMaxWidth()) {
                TabButton("Share link", Icons.Filled.Link, tab == "link", Modifier.weight(1f), purple) { tab = "link"; reset() }
                TabButton("Username", Icons.Filled.Person, tab == "username", Modifier.weight(1f), purple) { tab = "username"; reset() }
            }

            Spacer(Modifier.size(16.dp))

            Column(modifier = Modifier.heightIn(min = 220.dp)) {
                // Mode picker — brand-color dropdown with glyph.
                Text("MODE", fontSize = 10.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.textMuted)
                Spacer(Modifier.size(4.dp))
                Box {
                    Row(
                        modifier = Modifier.fillMaxWidth()
                            .clip(RoundedCornerShape(10.dp))
                            .background(WTheme.surface)
                            .border(1.5.dp, card.accent, RoundedCornerShape(10.dp))
                            .clickable { modeOpen = true }
                            .padding(horizontal = 12.dp, vertical = 8.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Box(
                            Modifier.size(24.dp).clip(RoundedCornerShape(6.dp)).background(card.accent.copy(alpha = 0.10f)),
                            contentAlignment = Alignment.Center,
                        ) { ModeGlyph(card, card.accent, 13.sp, 16.dp) }
                        Spacer(Modifier.width(8.dp))
                        Text(card.title, fontSize = 14.sp, fontWeight = FontWeight.Black, color = card.accent, modifier = Modifier.weight(1f))
                        Icon(Icons.Filled.ExpandMore, null, tint = card.accent, modifier = Modifier.size(18.dp))
                    }
                    DropdownMenu(expanded = modeOpen, onDismissRequest = { modeOpen = false }) {
                        modes.forEach { m ->
                            DropdownMenuItem(
                                onClick = { card = m; modeOpen = false; reset() },
                                text = {
                                    Row(verticalAlignment = Alignment.CenterVertically) {
                                        Box(
                                            Modifier.size(24.dp).clip(RoundedCornerShape(6.dp)).background(m.accent.copy(alpha = 0.10f)),
                                            contentAlignment = Alignment.Center,
                                        ) { ModeGlyph(m, m.accent, 13.sp, 16.dp) }
                                        Spacer(Modifier.width(8.dp))
                                        Text(m.title, fontSize = 14.sp, fontWeight = FontWeight.Black, color = m.accent, modifier = Modifier.weight(1f))
                                        if (m.id == card.id) Icon(Icons.Filled.Check, null, tint = m.accent, modifier = Modifier.size(14.dp))
                                    }
                                },
                            )
                        }
                    }
                }

                Spacer(Modifier.size(12.dp))

                if (tab == "link") {
                    val url = inviteUrl
                    if (url == null) {
                        PrimaryButton(if (busy) "Creating…" else "Generate invite link", purple, !busy) {
                            busy = true; error = null; copied = false
                            scope.launch {
                                val r = InviteService.createInvite(card.engineMode!!.name, null)
                                busy = false
                                if (r.code != null) inviteUrl = "https://wordocious.com/vs/join/${r.code}"
                                else error = r.error ?: "Failed to create invite"
                            }
                        }
                    } else {
                        Row(
                            modifier = Modifier.fillMaxWidth()
                                .clip(RoundedCornerShape(10.dp)).background(WTheme.bg)
                                .border(1.5.dp, WTheme.border, RoundedCornerShape(10.dp))
                                .padding(horizontal = 12.dp, vertical = 10.dp),
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Text(url, fontSize = 11.sp, fontWeight = FontWeight.Bold, color = WTheme.text,
                                maxLines = 1, overflow = TextOverflow.Ellipsis, modifier = Modifier.weight(1f))
                            Box(
                                Modifier.size(28.dp).clip(RoundedCornerShape(6.dp))
                                    .background(WTheme.surface).border(1.5.dp, WTheme.border, RoundedCornerShape(6.dp))
                                    .clickable { clipboard.setText(AnnotatedString(url)); copied = true },
                                contentAlignment = Alignment.Center,
                            ) {
                                Icon(if (copied) Icons.Filled.Check else Icons.Filled.ContentCopy, "Copy",
                                    tint = if (copied) Color(0xFF16A34A) else purple, modifier = Modifier.size(14.dp))
                            }
                        }
                        Spacer(Modifier.size(12.dp))
                        PrimaryButton("Share", purple, true, Icons.Filled.Share) {
                            val text = "Come play me on Wordocious — ${card.title}."
                            val intent = Intent(Intent.ACTION_SEND).apply {
                                type = "text/plain"
                                putExtra(Intent.EXTRA_TEXT, "$text\n$url")
                            }
                            context.startActivity(Intent.createChooser(intent, "Invite a friend").apply {
                                addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                            })
                        }
                        Spacer(Modifier.size(8.dp))
                        Text("Link expires in 24 hours.", fontSize = 10.sp, fontWeight = FontWeight.Bold,
                            color = WTheme.textMuted, modifier = Modifier.fillMaxWidth(),
                            textAlign = androidx.compose.ui.text.style.TextAlign.Center)
                    }
                } else {
                    val sent = sentTo
                    if (sent == null) {
                        Text("USERNAME", fontSize = 10.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.textMuted)
                        Spacer(Modifier.size(4.dp))
                        OutlinedTextField(
                            value = username, onValueChange = { username = it },
                            placeholder = { Text("e.g. wordmaster", fontSize = 14.sp) },
                            singleLine = true,
                            keyboardOptions = KeyboardOptions(autoCorrectEnabled = false),
                            modifier = Modifier.fillMaxWidth(),
                        )
                        Spacer(Modifier.size(12.dp))
                        PrimaryButton(if (busy) "Sending…" else "Send invite", purple, !busy) {
                            val clean = username.trim().trimStart('@')
                            if (clean.isEmpty()) { error = "Enter a username"; return@PrimaryButton }
                            busy = true; error = null
                            scope.launch {
                                val r = InviteService.createInvite(card.engineMode!!.name, clean)
                                busy = false
                                if (r.code != null) sentTo = clean else error = r.error ?: "Failed to send invite"
                            }
                        }
                    } else {
                        Column(
                            modifier = Modifier.fillMaxWidth()
                                .clip(RoundedCornerShape(12.dp))
                                .background(Color(0xFFF0FDF4))
                                .border(1.5.dp, Color(0xFF86EFAC), RoundedCornerShape(12.dp))
                                .padding(16.dp),
                            horizontalAlignment = Alignment.CenterHorizontally,
                        ) {
                            Icon(Icons.Filled.Check, null, tint = Color(0xFF16A34A), modifier = Modifier.size(24.dp))
                            Spacer(Modifier.size(6.dp))
                            Text("Invite sent to @$sent", fontSize = 14.sp, fontWeight = FontWeight.Black, color = Color(0xFF166534))
                            Text("They'll see it the next time they open Wordocious.",
                                fontSize = 10.sp, fontWeight = FontWeight.Bold, color = Color(0xFF16A34A),
                                modifier = Modifier.padding(top = 4.dp))
                        }
                    }
                }

                error?.let {
                    Spacer(Modifier.size(12.dp))
                    Text(it, fontSize = 12.sp, fontWeight = FontWeight.Bold, color = Color(0xFFEF4444),
                        modifier = Modifier.fillMaxWidth(), textAlign = androidx.compose.ui.text.style.TextAlign.Center)
                }
            }
        }
    }
}

@Composable
private fun TabButton(
    label: String, icon: androidx.compose.ui.graphics.vector.ImageVector,
    active: Boolean, modifier: Modifier, accent: Color, onClick: () -> Unit,
) {
    Row(
        modifier = modifier
            .clip(RoundedCornerShape(12.dp))
            .background(if (active) accent else WTheme.bg)
            .border(1.5.dp, if (active) accent else WTheme.border, RoundedCornerShape(12.dp))
            .clickable { onClick() }
            .padding(vertical = 9.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, null, tint = if (active) Color.White else WTheme.textMuted, modifier = Modifier.size(14.dp))
        Spacer(Modifier.width(6.dp))
        Text(label, fontSize = 12.sp, fontWeight = FontWeight.ExtraBold, color = if (active) Color.White else WTheme.textMuted)
    }
}

@Composable
private fun PrimaryButton(
    label: String, color: Color, enabled: Boolean,
    icon: androidx.compose.ui.graphics.vector.ImageVector? = null, onClick: () -> Unit,
) {
    Row(
        modifier = Modifier.fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(if (enabled) color else color.copy(alpha = 0.5f))
            .clickable(enabled = enabled) { onClick() }
            .padding(vertical = 12.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (icon != null) { Icon(icon, null, tint = Color.White, modifier = Modifier.size(16.dp)); Spacer(Modifier.width(6.dp)) }
        Text(label, fontSize = 14.sp, fontWeight = FontWeight.Black, color = Color.White)
    }
}
